# Generated by Django 4.2.23 on 2025-08-14 08:09

import django.contrib.auth.models
import django.core.validators
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('recipes', '0002_alter_foodgramuser_username'),
    ]

    operations = [
        migrations.AlterModelOptions(
            name='favorite',
            options={'default_related_name': 'favorites', 'ordering': ('user', 'recipe'), 'verbose_name': 'избранный рецепт', 'verbose_name_plural': 'Избранные рецепты'},
        ),
        migrations.AlterModelOptions(
            name='shoppingcart',
            options={'default_related_name': 'carts', 'ordering': ('user', 'recipe'), 'verbose_name': 'список покупок', 'verbose_name_plural': 'Списки покупок'},
        ),
        migrations.AlterModelManagers(
            name='foodgramuser',
            managers=[
                ('objects', django.contrib.auth.models.UserManager()),
            ],
        ),
        migrations.AlterField(
            model_name='foodgramuser',
            name='username',
            field=models.CharField(max_length=150, unique=True, validators=[django.core.validators.RegexValidator(message='Имя пользователя может содержать только буквы, цифры и символы @ . - _', regex='^[\\w.@-]+$')], verbose_name='Никнейм'),
        ),
        migrations.AddConstraint(
            model_name='favorite',
            constraint=models.UniqueConstraint(fields=('user', 'recipe'), name="'favorite's_unique_user_recipe"),
        ),
        migrations.AddConstraint(
            model_name='shoppingcart',
            constraint=models.UniqueConstraint(fields=('user', 'recipe'), name="'shoppingcart's_unique_user_recipe"),
        ),
    ]
